lnbits/core/services/users.py

```diff
@@ -0,0 +1,128 @@
+from pathlib import Path
+from typing import Optional
+from uuid import UUID, uuid4
+
+from loguru import logger
+
+from lnbits.core.extensions.models import UserExtension
+from lnbits.settings import (
+    EditableSettings,
+    SuperSettings,
+    send_admin_user_to_saas,
+    settings,
+)
+
+from ..crud import (
+    create_account,
+    create_admin_settings,
+    create_wallet,
+    get_account,
+    get_account_by_email,
+    get_account_by_pubkey,
+    get_account_by_username,
+    get_super_settings,
+    get_user_from_account,
+    update_super_user,
+    update_user_extension,
+)
+from ..helpers import to_valid_user_id
+from ..models import (
+    Account,
+    User,
+    UserExtra,
+)
+from .settings import update_cached_settings
+
+
+async def create_user_account(
+    account: Optional[Account] = None, wallet_name: Optional[str] = None
+) -> User:
+    if not settings.new_accounts_allowed:
+        raise ValueError("Account creation is disabled.")
+    if account:
+        if account.username and await get_account_by_username(account.username):
+            raise ValueError("Username already exists.")
+
+        if account.email and await get_account_by_email(account.email):
+            raise ValueError("Email already exists.")
+
+        if account.pubkey and await get_account_by_pubkey(account.pubkey):
+            raise ValueError("Pubkey already exists.")
+
+        if account.id:
+            user_uuid4 = UUID(hex=account.id, version=4)
+            assert user_uuid4.hex == account.id, "User ID is not valid UUID4 hex string"
+        else:
+            account.id = uuid4().hex
+
+    account = await create_account(account)
+    await create_wallet(
+        user_id=account.id,
+        wallet_name=wallet_name or settings.lnbits_default_wallet_name,
+    )
+
+    for ext_id in settings.lnbits_user_default_extensions:
+        user_ext = UserExtension(user=account.id, extension=ext_id, active=True)
+        await update_user_extension(user_ext)
+
+    user = await get_user_from_account(account)
+    assert user, "Cannot find user for account."
+
+    return user
+
+
+async def check_admin_settings():
+    if settings.super_user:
+        settings.super_user = to_valid_user_id(settings.super_user).hex
+
+    if settings.lnbits_admin_ui:
```

Comment:
could be if not, do nothing

Reply:
I've just moved code, no other type of refactoring
can be addressed in different PR